engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -843,72 +846,103 @@ protected boolean handleDisconnectWithInvestigation(final AgentAttache attache,
                 s_logger.debug("Caught exception while getting agent's next status", ne);
             }
 
+            // For log and alert purposes later
+            final DataCenterVO dcVO = _dcDao.findById(host.getDataCenterId());
+            final HostPodVO podVO = _podDao.findById(host.getPodId());
+            final String hostDesc = "[name: " + host.getName() + " (id:" + host.getId() + "), availability zone: " + dcVO.getName() + ", pod: " + podVO.getName() + "]";
+            final String hostShortDesc = "Host " + host.getName() + " (id:" + host.getId() + ")";
+
+            final ResourceState resourceState = host.getResourceState();
+            if (resourceState == ResourceState.Disabled || resourceState == ResourceState.Maintenance || resourceState == ResourceState.ErrorInMaintenance) {
+                // If we are in this resourceState, no need to investigate or do anything.  AgentMonitor will handle when in these resourceStates
+                s_logger.info(hostShortDesc + " has disconnected with event " + event + ",  but is in Resource State of " + resourceState + ", so doing nothing");
+                return true;
+            }
+
             if (nextStatus == Status.Alert) {
-                /* OK, we are going to the bad status, let's see what happened */
-                s_logger.info("Investigating why host " + hostId + " has disconnected with event " + event);
+                /* Our next Agent transition state is Alert
+                 * Let's see if the host down or why we had this event
+                 */
+                s_logger.info("Investigating why host " + hostShortDesc + " has disconnected with event " + event);
```

Comment:
👍 good improvement, but though it is only (a comment and) a log statement, this entails an interface of the system. the ecosystem may query logs for the text and no longer find the hostId thus not being able to take mitigating actions any more. I'd rather see a less destructive change like 'hostId + " (" + hostShortDesc + ") "'

We may get away with it but it does require extensive testing by the whole community :/.

Reply:
@DaanHoogland good thought, I didn’t think about that.  However, the hostShortDesc does include the hostId as part of it.  So maybe it’s ok?

final String hostShortDesc = "Host " + host.getName() + " (id:" + host.getId() + ")";

Reply:
yeah, maybe. intesive testing still required...